packages/next/server/web/adapter.ts

```diff
@@ -180,6 +186,15 @@ export function blockUnallowedResponse(
   promise: Promise<FetchEventResult>
 ): Promise<FetchEventResult> {
   return promise.then((result) => {
+    const { nextUrl } = ResponseToRequest.get(result.response) || {}
```

Comment:
**suggestion:** while this works fine, we could achieve the same behavior, without introducing a weakmap (kind of side-effect-ish), by returning nextUrl as part of `FetchEventResult`